// + Uncaught exception handler
process.on('uncaughtException', (err) => {
    console.log('UNCAUGHT EXCEPTION! ðŸ’¥ Shutting down...');
    console.log(err.name, err.message);
    process.exit(1);
});

//** Start server
const app = require('./app');
const port = 3000;
const server = app.listen(port, () => {
    console.log(`Server is running on port ${port}`);
});

//** Unhandled promise rejection
process.on('unhandledRejection', (err) => {
    console.log('UNHANDLED REJECTION! ðŸ’¥ Shutting down...');
    console.log(err.name, err.message);
    server.close(() => {
        process.exit(1);
    });
});
